src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -360,9 +421,97 @@ public void simInit(PhysicsSim sim) {
+				networkTableDrivebase.getDoubleTopic("Front right actual velocity").publish();
+		backLeftActualVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Back left actual velocity").publish();
+		backRightActualVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Back right actual velocity").publish();
+
+		frontLeftTargetVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Front left target velocity").publish();
+		frontRightTargetVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Front right target velocity").publish();
+		backLeftTargetVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Back left target velocity").publish();
+		backRightTargetVelocityPublisher =
+				networkTableDrivebase.getDoubleTopic("Back right target velocity").publish();
+
+		frontLeftActualAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Front left actual angle").publish();
+		frontRightActualAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Front right actual angle").publish();
+		backLeftActualAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Back left actual angle").publish();
+		backRightActualAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Back right actual angle").publish();
+
+		frontLeftTargetAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Front left target angle").publish();
+		frontRightTargetAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Front right target angle").publish();
+		backLeftTargetAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Back left target angle").publish();
+		backRightTargetAnglePublisher =
+				networkTableDrivebase.getDoubleTopic("Back right target angle").publish();
+
+		frontLeftActualVelocityPublisher.set(0.0);
+		frontRightActualVelocityPublisher.set(0.0);
+		backLeftActualVelocityPublisher.set(0.0);
+		backRightActualVelocityPublisher.set(0.0);
+
+		frontLeftTargetVelocityPublisher.set(0.0);
+		frontRightTargetVelocityPublisher.set(0.0);
+		backLeftTargetVelocityPublisher.set(0.0);
+		backRightTargetVelocityPublisher.set(0.0);
+
+		frontLeftActualAnglePublisher.set(0.0);
+		frontRightActualAnglePublisher.set(0.0);
+		backLeftActualAnglePublisher.set(0.0);
+		backRightActualAnglePublisher.set(0.0);
+
+		frontLeftTargetAnglePublisher.set(0.0);
+		frontRightTargetAnglePublisher.set(0.0);
+		backLeftTargetAnglePublisher.set(0.0);
+		backRightTargetAnglePublisher.set(0.0);
+
+		SmartDashboard.putData("Translational PID", compTranslationalPID);
+		SmartDashboard.putData("Rotational PID", compRotationalPID);
+	}
+
+	private double oldTranslationalSetpoint = 0.0;
+	private double oldRotationalSetpoint = 0.0;
+
 	@Override
 	public void periodic() {
 		pose = poseEstimator.update(gyroscope.getAngle(), getModulePositions());
 		field.setRobotPose(pose);
+
+		for (MotorController motor : moduleDriveMotors) {
+			if (compTranslationalPID.getSetpoint() != oldTranslationalSetpoint) {
```

Comment:
Not being that familiar with the drive base code, I am having a hard time wrapping my head around why the conditional check is not outside the loop.
Seems like with the conditional inside the loop there is a possibility that not all motors will get updated in each periodic.

Reply:
Ah - that might explain why PID changes didn't seem to have much effect